src/prefect/cli/deployment.py

```diff
@@ -557,6 +563,14 @@ async def build(
 
     deployment_loc = output_file or f"{obj_name}-deployment.yaml"
     await deployment.to_yaml(deployment_loc)
-    exit_with_success(
-        f"Deployment YAML created at '{Path(deployment_loc).absolute()!s}'."
-    )
+
+    if _apply:
+        deployment_id = await deployment.apply()
+        app.console.print(
+            f"Deployment '{deployment.flow_name}/{deployment.name}' successfully created with id '{deployment_id}'.",
+            style="green",
+        )
+    else:
+        exit_with_success(
```

Comment:
Since we're still creating the .yaml, should we still output this message or some form of it so a user isn't surprised about a 'random' yaml file laying around?

Reply:
Good call, will update